WordPress/src/main/java/org/wordpress/android/ui/WPWebViewActivity.java

```diff
@@ -275,15 +276,27 @@ protected void loadContent() {
         String username = extras.getString(AUTHENTICATION_USER, "");
         String password = extras.getString(AUTHENTICATION_PASSWD, "");
         String authURL = extras.getString(AUTHENTICATION_URL);
-        if (extras.getBoolean(USE_GLOBAL_WPCOM_USER, false)) {
-            username = mAccountStore.getAccount().getUserName();
-        }
 
         if (TextUtils.isEmpty(addressToLoad) || !UrlUtils.isValidUrlAndHostNotNull(addressToLoad)) {
             AppLog.e(AppLog.T.UTILS, "Empty or null or invalid URL passed to WPWebViewActivity");
             Toast.makeText(this, getText(R.string.invalid_site_url_message),
                     Toast.LENGTH_SHORT).show();
             finish();
+            return;
+        }
+
+        if (extras.getBoolean(USE_GLOBAL_WPCOM_USER, false)) {
+            username = mAccountStore.getAccount().getUserName();
+
+            // Custom domains are not properly authenticated due to a server side(?) issue, so this gets around that
+            List<SiteModel> wpComSites = mSiteStore.getWPComSites();
```

Comment:
Fixed in cb6442b9b21d3783c4304816455534e9bb6ab08f

I've also considered adding a check to see if the url actually changed and breaking the loop if it did since we don't need to go through other sites. However, I feel that's almost as expensive as completing the loop and it will also add some extra code, so probably not worth it. What do you think?